core/src/main/java/org/elasticsearch/index/mapper/core/MapperTestUtils.java

```diff
@@ -44,37 +45,50 @@
 public class MapperTestUtils {
 
     public static MapperService newMapperService(Path tempDir, Settings indexSettings) {
-        return newMapperService(new Index("test"), Settings.builder()
-                .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-                .put("path.home", tempDir)
-                .put(indexSettings)
-                .build());
+        return newMapperService(tempDir, indexSettings, Version.CURRENT);
+
+    }
+
+    public static MapperService newMapperService(Path tempDir, Settings indexSettings, Version version) {
```

Comment:
Thank you for moving this file into core! It will make unit testing mappers much easier!

I don't see why we need to make this complicated with a bunch of variants to create a mapper service. At least can we have just 2? Simple (takes a temp dir and index settings), and complicated that passes in IndicesModule (already constructed and any types added, or you could also take null to mean "create it") and version.